src/components/radio-button/radio-button-group/radio-button-group.test.tsx

```diff
@@ -0,0 +1,292 @@
+
+  expect(radioButton).toBeInTheDocument();
+  expect(screen.getByTestId("radio-svg")).toBeVisible();
+  expect(screen.getByText("Radio Button 1")).toBeVisible();
+});
+
+test("renders with non-`RadioButton` children when passed as an array", () => {
+  render(
+    <RadioButtonGroup name="group" onChange={() => {}}>
+      {[
+        <RadioButton key="radio1" value="radio1" label="Radio Button 1" />,
+        null,
+        undefined,
+        "foo",
+        <RadioButton key="radio2" value="radio2" label="Radio Button 2" />,
+      ]}
+    </RadioButtonGroup>
+  );
+
+  expect(screen.getAllByRole("radio")).toHaveLength(2);
+  expect(screen.getByText("foo")).toBeVisible();
+});
+
+test("renders fieldset with provided `legend`", () => {
+  render(
+    <RadioButtonGroup
+      name="group"
+      legend="Radio Group Legend"
+      onChange={() => {}}
+    >
+      <RadioButton value="radio1" label="Radio Button 1" />
+    </RadioButtonGroup>
+  );
+
+  const fieldset = screen.getByRole("group", { name: "Radio Group Legend" });
+
+  expect(fieldset).toBeVisible();
+});
+
+test("checks the correct `RadioButton` child when a value is passed", () => {
+  render(
+    <RadioButtonGroup name="group" value="radio2" onChange={() => {}}>
+      <RadioButton value="radio1" label="Radio Button 1" />
+      <RadioButton value="radio2" label="Radio Button 2" />
+    </RadioButtonGroup>
+  );
+
+  const radio1 = screen.getByRole("radio", { name: "Radio Button 1" });
+  const radio2 = screen.getByRole("radio", { name: "Radio Button 2" });
+
+  expect(radio1).not.toBeChecked();
+  expect(radio2).toBeChecked();
+});
+
+test("calls `onChange` when a `RadioButton` child is checked", async () => {
+  const user = userEvent.setup();
+  const onChange = jest.fn();
+  render(
+    <RadioButtonGroup name="group" onChange={onChange}>
+      <RadioButton value="radio1" label="Radio Button 1" />
+    </RadioButtonGroup>
+  );
+
+  const radioButton = screen.getByRole("radio");
+
+  await user.click(radioButton);
+
+  expect(onChange).toHaveBeenCalled();
+});
+
+test("calls `onBlur` when a `RadioButton` child is blurred", async () => {
+  const user = userEvent.setup();
+  const onBlur = jest.fn();
+  render(
+    <RadioButtonGroup name="group" onChange={() => {}} onBlur={onBlur}>
+      <RadioButton value="radio1" label="Radio Button 1" />
+    </RadioButtonGroup>
+  );
+
+  await user.tab();
+  await user.tab();
+
+  expect(onBlur).toHaveBeenCalled();
+});
+
+test("renders required `RadioButton` children when `required` prop is true", () => {
+  render(
+    <RadioButtonGroup name="group" required onChange={() => {}}>
+      <RadioButton value="radio1" label="Radio Button 1" />
+    </RadioButtonGroup>
+  );
+
+  const radioButton = screen.getByRole("radio", { name: "Radio Button 1" });
+
+  expect(radioButton).toBeRequired();
+});
+
+test("renders with inline legend when screen is larger than `adaptiveLegendBreakpoint`", () => {
```

Comment:
**suggestion**: I'd feel more comfortable if there was also a test for the screen being smaller than `adaptiveLegendBreakpoint`. I think this might also avoid the need for some of the tests below that you've marked as being only for coverage (although I'm not sure about this)